internal/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -193,44 +267,46 @@ func (q *DBJobQueue) Dequeue(ctx context.Context, jobTypes []string, channels []
 		return uuid.Nil, uuid.Nil, nil, "", nil, jobqueue.ErrDequeueTimeout
 	}
 
-	conn, err := q.pool.Acquire(ctx)
-	if err != nil {
-		return uuid.Nil, uuid.Nil, nil, "", nil, fmt.Errorf("error connecting to database: %v", err)
-	}
+	// add ourselves as a dequeuer
+	q.dequeuersLock.Lock()
+	c := make(chan struct{}, 1)
```

Comment:
You don't need to lock the call to `make()`. I think it might be easier to follow the code if you pull out
```
q.dequeuersLock.Lock()
defer q.dequeuersLock.Unlock()
return q.dequeuers.PushBack(c)
```
into a helper function. And the same with all the other places we hold `dequeuersLock`.

Reply:
Indeed, I created a helper struct, thanks, great suggestion.